packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -302,6 +328,7 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
             readonlyChangeHandler: (readonly?: boolean) => safeRaiseEvent(this, this.logger, "readonly", readonly),
         };
 
+        this.lastConnectionMode = this.client.mode;
```

Comment:
I'd rather make this.lastConnectionMode optional and have undefined as default value? This value has only indirect relationship to actual mode we will get on first connection, and (due to bug in the code or maybe by design) we can log it before connection happens. It's much more correct to see undefined than some value that is not related to actual connection modes.

Reply:
if we take the _connectionProps approach, maybe I don't need lastConnectionMode as a separate variable?